lib/storage.js

```diff
@@ -53,6 +54,26 @@ async function getAsset(assetReference, directory, name, disableDownload) {
         const source = new Asset(assetReference, directory, name);
         if (disableDownload) {
             console.log(`Skipping source file download for ${AssetComputeLogUtils.redactUrl(source.url)}`);
+            const protocol = URL.parse(assetReference.url).protocol;
+            if (protocol === DATA_PROTOCOL) {
+                await datauri.download(assetReference, source.path);
+                if(!fileExistsAndIsNotEmpty(source.path)){
+                    throw new SourceUnsupportedError(`Invalid or missing local file ${source.path}`);
+                }
+                let attempt = 0;
+                while(attempt <= MAX_RETRY_ATTEMPTS){
```

Comment:
nit pick: I think the retry logic should be abstracted inside `getPresignedUrl` (or even inside the temp cloud storage)

also, why 5 attempts?

Reply:
IMO, I would limit it to 3 retries. If it's a network/flakiness issue, 3 retries should be enough (1 retry usually works). Since each try uploads a file, it could be expensive to try this 5 times on a corrupt file or large file that times out uploading

Reply:
`5` is a randomly chosen number. Goal is to have retries, but not too many of them, and certainly not forever/until it eventually works.

Reply:
Reduce the count to 3, and moved the retry logic to datauri